lemminx-liberty/src/test/java/io/openliberty/LibertyDiagnosticTest.java

```diff
@@ -750,4 +750,115 @@ public void testUniquenessForNameChangedFeatureDiagnostic() throws BadLocationEx
+
+        XMLAssert.testDiagnosticsFor(serverXML, null, null, serverXMLURI,
+                invalid1);
+        //expecting code action to show all javaee platforms ignoring input case
+        List<String> featuresStartWithJavaEE = new ArrayList<>();
+        featuresStartWithJavaEE.add("javaee-6.0");
+        featuresStartWithJavaEE.add("javaee-7.0");
+        featuresStartWithJavaEE.add("javaee-8.0");
+        Collections.sort(featuresStartWithJavaEE);
+
+        List<CodeAction> codeActions = new ArrayList<>();
+        for (String nextFeature : featuresStartWithJavaEE) {
+            TextEdit texted = te(invalid1.getRange().getStart().getLine(), invalid1.getRange().getStart().getCharacter(),
+                    invalid1.getRange().getEnd().getLine(), invalid1.getRange().getEnd().getCharacter(), nextFeature);
+            CodeAction invalidCodeAction = ca(invalid1, texted);
+
+            codeActions.add(invalidCodeAction);
+        }
+
+        XMLAssert.testCodeActionsFor(serverXML, invalid1, codeActions.get(0), codeActions.get(1),
+                codeActions.get(2));
+
+        serverXML = String.join(newLine, //
+                "<server description=\"Sample Liberty server\">", //
+                "       <featureManager>", //
+                "               <feature>batch-1.0</feature>", //
+                "               <platform>ja</platform>", //
+                "               <platform>javaee-7.0</platform>", //
+                "       </featureManager>", //
+                "</server>" //
+        );
+        invalid1 = new Diagnostic();
+        invalid1.setRange(r(3, 25, 3, 27));
+        invalid1.setCode(LibertyDiagnosticParticipant.INCORRECT_PLATFORM_CODE);
+        invalid1.setMessage("ERROR: The platform \"ja\" does not exist.");
+
+        XMLAssert.testDiagnosticsFor(serverXML, null, null, serverXMLURI,
+                invalid1);
+        XMLAssert.testCodeActionsFor(serverXML, invalid1);
+        serverXML = String.join(newLine, //
+                "<server description=\"Sample Liberty server\">", //
+                "       <featureManager>", //
+                "               <feature>batch-1.0</feature>", //
+                "               <platform>miCroP</platform>", //
+                "               <platform>javaee-7.0</platform>", //
+                "       </featureManager>", //
+                "</server>" //
+        );
+        invalid1 = new Diagnostic();
+        invalid1.setRange(r(3, 25, 3, 31));
+        invalid1.setCode(LibertyDiagnosticParticipant.INCORRECT_PLATFORM_CODE);
+        invalid1.setMessage("ERROR: The platform \"miCroP\" does not exist.");
+
+        XMLAssert.testDiagnosticsFor(serverXML, null, null, serverXMLURI,
+                invalid1);
+        //  expecting code action to show only microprofile platforms since
+        //      1. javaee is already included
+        //      2. jakartaee is conflicting with javaee platforms
```

Comment:
These comments about javaee and jakartaee do not apply to the microprofile test. The previous test for `ja` had a comment that related to these two comments, because it used to say there would be no code actions because of 1 and 2.